edgelet/hsm-rs/src/crypto.rs

```diff
@@ -194,6 +194,30 @@ fn make_certification_props(props: &CertificateProperties) -> Result<CERT_PROPS_
             unsafe { cert_properties_destroy(handle) };
             ErrorKind::CertProps
         })?;
+
+    if props.san_entries.len() > 0 {
+        let result: Result<Vec<CString>, NulError> = props
+            .san_entries
+            .iter()
+            .map(|s: &String| CString::new(s.clone()))
```

Comment:
This `clone` call may not be needed. Looks like `CString::new` can work with `&str`.

Reply:
CString consumes so we need a clone.

Reply:
I think its safe to pass a `&str` for `CString::new`. There is still an alloc happening under the covers but with `clone` there are 2 allocs.